### ISSUE TITLE
One Str node per word disables multi-word rules (e.g. write-good passive voice, whitelist)

### ISSUE THREAD

**Issue by koppor** [textlint/textlint-plugin-latex2e#481]
The plugin emits one `Str` node per word (spaces become `Html` nodes), while the Markdown plugin emits one `Str` per text run. Rules that analyze a sentence inside a `Str` therefore never see more than one word in `.tex` files.

Repro with textlint 15.8, textlint-plugin-latex2e 1.2.1, textlint-rule-write-good 2.0.0 and `{"plugins":{"latex2e":true},"rules":{"write-good":true}}`:

```
The paper was written by us. There is the the problem.
```

- `t.md`: 3 problems (`"was written" may be passive voice`, `"There is" is unnecessary verbiage`, `"the" is repeated`)
- `t.tex`: 0 problems

The `Str` nodes of `You only report the measurements.` are `"You"`, `"only"`, `"report"`, `"the"`, `"measurements."`. This also exposed a write-good bug where a whitelisted word at index 0 is still reported (btford/write-good#146).

Expected: adjacent text and whitespace of a paragraph are merged into one `Str` node, as in the Markdown AST. I understand #470 and #463 may make this moot; filing it so the behavior is on record.

AI-generated (Claude Code).
